src/Enterspeed.Cli/Commands/Schema/ImportSchemaCommand.cs

```diff
@@ -96,27 +96,57 @@ private async Task<bool> UpdateExistingSchema(SchemaFile schemaFile, QueryMappin
                 return false;
             }
 
-            var updateSchemaResponse = await _mediator.Send(new UpdateMappingSchemaRequest
+            var relativeDirectoryPathInEnterspeed = Path.GetDirectoryName(existingSchema.Name.TrimEnd(Path.DirectorySeparatorChar));
+            var relativeDirectoryPathOnDisk = schemaFile.RelativeSchemaDirectory;
+
+            if (!relativeDirectoryPathOnDisk.Equals(relativeDirectoryPathInEnterspeed))
+            {
+                await UpdateSchemaName(existingSchema, relativeDirectoryPathOnDisk);
+            }
+
+            var updateMappingSchemaVersionResponse = await _mediator.Send(new UpdateMappingSchemaVersionRequest
             {
                 Format = schemaFile.Format,
-                MappingSchemaId = queryMappingSchemaResponse.Id.MappingSchemaGuid,
+                MappingSchemaId = matchingSchema.Id.MappingSchemaGuid,
                 Version = existingSchema.LatestVersion,
                 Schema = schemaFile.GetSchemaContent()
             });
 
-            _outputService.Write($"Successfully updated schema: {schemaFile.Alias} Version: {updateSchemaResponse.Version}");
+            _outputService.Write($"Successfully updated schema: {schemaFile.Alias} Version: {updateMappingSchemaVersionResponse.Version}");
 
             return true;
         }
 
+        private async Task UpdateSchemaName(GetMappingSchemaResponse existingSchema, string relativeDirectoryPathOnDisk)
+        {
+            var name = SchemaExtensions.BuildNewSchemaName(existingSchema.Name, relativeDirectoryPathOnDisk);
+            var updateSchemaResponse = await _mediator.Send(new UpdateMappingSchemaRequest()
+            {
+                Name = name,
+                MappingSchemaId = existingSchema.Version.Id.MappingSchemaGuid,
+            });
+
+            _outputService.Write($"Successfully updated name to: {name}");
+        }
+
         /// <summary>
         /// As the management API does not support upserts we first creates the empty schema and then updates the schema with content
         /// </summary>
         private async Task<bool> CreateNewSchema(SchemaFile schemaFile)
         {
+            string name;
+            if (!string.IsNullOrEmpty(schemaFile.RelativeSchemaDirectory))
+            {
+                name = Path.Combine(schemaFile.RelativeSchemaDirectory, schemaFile.Alias);
+            }
+            else
+            {
+                name = schemaFile.Alias;
+            }
```

Comment:
Should this be handled in the SchemaExtensions class or maybe moved to a SchemaNameService?